AtmosphereAutopilot/Modules/CruiseController.cs

```diff
@@ -547,7 +560,11 @@ protected override void _drawGUI(int id)
             {
                 if (height_mode == HeightMode.FlightPathAngle)
                 {
-                    desired_vertspeed.Value = (float) (vessel.horizontalSrfSpeed * Math.Tan(desired_vertspeed * dgr2rad));
+                    if (90.0f - Mathf.Abs(desired_vertspeed) < vert_angle_epsilon)
+                        // don't put a value close to infinity to the field
+                        desired_vertspeed.Value = 9999.0f * Mathf.Sign(desired_vertspeed);
+                    else
+                        desired_vertspeed.Value = (float)(vessel.horizontalSrfSpeed * Math.Tan(desired_vertspeed * dgr2rad));
```

Comment:
I feel like a lot of this would look better if we didn't use desired_vertspeed for both vertical speed and angle modes.  Maybe, split it in two?

Reply:
Splitting it means another field would be added to the GUI. That might not look well if more vertical modes are added. Anyway I'll give it a shot.

Reply:
It's not autogui-ed, so you can add a simple if and render one of the two fields depending on the mode.

Reply:
https://github.com/T2Fat2Fly/AtmosphereAutopilot/blob/6a51f2d149c11e96940d17c2c7d97a37d0f94b43/AtmosphereAutopilot/Modules/CruiseController.cs#L223

You mean like [this](https://github.com/T2Fat2Fly/AtmosphereAutopilot/commit/6a51f2d149c11e96940d17c2c7d97a37d0f94b43)?
That doesn't look more elegant IMO. Plus the "edit setpoint before activation" feature is lost since the two setpoints are now out of sync. Syncing them back off is basically equal to using a single field variable.

Reply:
> Plus the "edit setpoint before activation" feature is lost since the two setpoints are now out of sync. Syncing them back off is basically equal to using a single field variable.

A self-inflicted pain, caused by the previous decision to only show one of the two at the time =).
The idea to sync them is also causing a lot of paint with 9999 and angle-to-vertspeed conversion.
Additional space can be saved by rotating desired_direction to planet2vesNorm by using Vector3.RotateTowards instead of Math.Tan and fpa_angle_epsilon stuff.

I'll be content with the current version of the code in PR with singe variable, renamed from desired_vertspeed to desired_vertsetpoint to better represent it's meaning, but I still suggest showing both v/s and fpa and getting rid of all the conversions, syncs and epsilons.